packages/block-library/src/search/index.php

```diff
@@ -78,27 +95,40 @@ function render_block_core_search( $attributes ) {
 		}
 
 		$button_markup = sprintf(
-			'<button type="submit" class="wp-block-search__button">%s</button>',
+			'<button type="submit"%1$s%2$s>%3$s</button>',
+			sprintf( ' class="%s"', esc_attr( implode( ' ', $button_classes ) ) ),
+			( ! empty( $button_styles ) ) ? sprintf( ' style="%s"', esc_attr( $button_styles ) ) : '',
 			$button_internal_markup
 		);
 	}
 
+	$field_classes = array_merge(
+		array( 'wp-block-search__inside-wrapper' ),
+		( array_key_exists( 'buttonPosition', $attributes ) && 'button-inside' === $attributes['buttonPosition'] )
+			? array_filter( $shared_classes )
+			: array()
+	);
+	$field_styles = ( array_key_exists( 'buttonPosition', $attributes ) && 'button-inside' === $attributes['buttonPosition'] )
+		? $shared_styles
+		: '';
+
 	if ( ! empty( $attributes['width'] ) && ! empty( $attributes['widthUnit'] ) ) {
 		if ( ! empty( $attributes['buttonPosition'] ) && 'button-only' !== $attributes['buttonPosition'] ) {
-			$width_styles = ' style="width: ' . $attributes['width'] . $attributes['widthUnit'] . ';"';
+			$field_styles .= ' width: ' . esc_attr( $attributes['width'] ) . esc_attr( $attributes['widthUnit'] ) . ';"';
```

Comment:
We have an extraneous `"` character in this markup.
```suggestion
			$field_styles .= ' width: ' . esc_attr( $attributes['width'] ) . esc_attr( $attributes['widthUnit'] ) . ';';
```